src/tools/archiveTools.cpp

```diff
@@ -125,6 +125,30 @@ unsigned int getArchiveMediaCount(const zim::Archive& archive) {
   return counter;
 }
 
+unsigned int getArchiveArticleCount(const zim::Archive& archive) {
+  // [HACK]
+  // getArticleCount() returns different things depending of the "version" of the zim.
+  // On old zim (<=6), it returns the number of entry in `A` namespace
+  // On recent zim (>=7), it returns:
+  //  - the number of entry in `C` namespace (==getEntryCount) if no frontArticleIndex is present
+  //  - the number of front article if a frontArticleIndex is present
+  // The use case >=7 without frontArticleIndex is pretty rare so we don't care
```

Comment:
Nice poetry!

On a more serious note, presence of the `frontArticleIndex` doesn't guarantee that the return value of `zim::Archive::getArticleCount()` is meaningful. It will be equal to zero for ZIM files created by tools linked against pre-7.2.0 libzim7 which weren't upgraded to supply the `FRONT_ARTICLE` hint (this was fixed by https://github.com/openzim/libzim/commit/682e38b0 in openzim/libzim#647). Maybe `kiwix::getArchiveArticleCount()` should return the number of entries in `C` namespace in that case?

Reply:
Those zim files are pretty rare I think. mwoffliner is still using the old libzim version as [node-libzim](https://github.com/openzim/node-libzim) is not yet ported (openzim/node-libzim#72) and we've created very few zim with the python wrapper (if we publish them). @kelson42 and @rgaudin may confirm this.

`zim::Archive::getArticleCount()` already return the number of entries in `C` namespace if there is no front article list. (https://github.com/openzim/libzim/blob/master/src/archive.cpp#L69-L78)